Document Kafka Source Schema management

eolivelli opened the pull request:
Add documentation about Kafka Source manages Schema information.

This patch documents the improvements implemented with: 
#9448 - KafkaSource - allow to manage Avro Encoded messages
#10002 - KafkaSource - Implement KeyValue support for KafkaBytesSource

Jennifer88huang commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
 
+### Schema Management
+
+This Kafka Source connector applies the Schema to the topic depennding on the data type that is present on the Kafka topic.
```
depending

eolivelli replied:
done

Jennifer88huang commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
+### Schema Management
+
+This Kafka Source connector applies the Schema to the topic depennding on the data type that is present on the Kafka topic.
+We are detecting the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
```
```suggestion
You can detect the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
```

Jennifer88huang commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
+This Kafka Source connector applies the Schema to the topic depennding on the data type that is present on the Kafka topic.
+We are detecting the data type from the `keyDeserializationClass` and `valueDeserializationClass` configuration parameters.
+
+For instance if the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer` then we set Schema.STRING() as schema type
+on the Pulsar topic.
+
```
```suggestion
If the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer`, you can set Schema.STRING() as schema type on the Pulsar topic.

```

Jennifer88huang commented on site2/docs/io-kafka-source.md:
````diff
@@ -95,7 +123,7 @@ Here is an example of using the Kafka source connecter with the configuration fi
    
    ```bash
-   $ docker pull apachepulsar/pulsar:2.4.0
+   $ docker pull apachepulsar/pulsar:2.8.0
````
If the version should be updated with release, maybe we can use variable
`{{pulsar:version}}`

eolivelli replied:
thanks I didn't find the variable

Jennifer88huang reviewed (commented):
Good job, @eolivelli 
I have one minor suggestion on the writing: when writing technical docs, try to use second person "you" and present tense. 
For guides, refer to https://developers.google.com/style/person

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
+For instance if the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer` then we set Schema.STRING() as schema type
+on the Pulsar topic.
+
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` we download the AVRO schema from the Confluent Schema Registry®
+and we set it properly on the Pulsar topic. In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
+of the source.
```
Avoid the first-person pronouns `I` and `we`. Use the second-person pronoun `you` as much as possible.

eolivelli replied:
got it

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
+and we set it properly on the Pulsar topic. In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
+of the source.
+
+When `keyDeserializationClass` has any value that is different from the default value of `org.apache.kafka.common.serialization.StringDeserializer` it means 
+that you do not have a String as a key, in this case the Kafka Source uses the KeyValue schema type with the SEPARATED encoding.
```
Break up long sentences into short ones can improve readability.

eolivelli replied:
done

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,47 @@ The configuration of the Kafka source connector has the following properties.
+
+Topic compaction and partition routing use the Pulsar key, that contains the Kafka key, and so they are driven by the same value that you have on Kafka.
+
+When you are consuming data from the Pulsar topic you can use the KeyValue schema, this way you can decode properly the data.
```
```suggestion
When you consume data from Pulsar topics, you can use the `KeyValue` schema. In this way, you can decode the data properly.
```

Anonymitaet reviewed (commented):
Thanks for your contribution. I've left some comments, pls check overall.

eolivelli commented:
@Jennifer88huang @Anonymitaet I have addressed your comments and tested again the rendering on my local env.
PTAL

it is very slow to generate the website, I am using
```
yarn build
yarn start
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
 
+### Schema Management
+
+This Kafka Source connector applies the Schema to the topic depending on the data type that is present on the Kafka topic.
```
```suggestion
This Kafka source connector applies the schema to the topic depending on the data type that is present on the Kafka topic.
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+
+If the `valueDeserializationClass` is `org.apache.kafka.common.serialization.StringDeserializer`, you can set Schema.STRING() as schema type on the Pulsar topic.
+
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` Pulsar downloads the AVRO schema from the Confluent Schema Registry®
```
```suggestion
If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer`, Pulsar downloads the AVRO schema from the Confluent Schema Registry®
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+If `valueDeserializationClass` is `io.confluent.kafka.serializers.KafkaAvroDeserializer` Pulsar downloads the AVRO schema from the Confluent Schema Registry®
+and set it properly on the Pulsar topic.
+
+In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
```
```suggestion
In this case, you need to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+In this case you have to set `schema.registry.url` inside of the `consumerConfigProperties` configuration entry
+of the source.
+
+When `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer` it means 
```
```suggestion
If `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer`, it means 
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+When `keyDeserializationClass` is not `org.apache.kafka.common.serialization.StringDeserializer` it means 
+that you do not have a String as key and the Kafka Source uses the KeyValue schema type with the SEPARATED encoding.
+
+We are also supporting AVRO format for keys as well.
```
```suggestion
Pulsar supports AVRO format for keys.
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+
+We are also supporting AVRO format for keys as well.
+
+This way on Pulsar you have a topic with these properties:
```
```suggestion
In this case, you can have a Pulsar topic with the following properties:
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+- Schema: KeyValue schema with SEPARATED encoding
+- Key: the content of key of the Kafka message (base64 encoded)
+- Value: the content of value of the Kafka message
+- KeySchema: the schema, detected from `keyDeserializationClass`
```
```suggestion
- KeySchema: the schema detected from `keyDeserializationClass`
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+- Key: the content of key of the Kafka message (base64 encoded)
+- Value: the content of value of the Kafka message
+- KeySchema: the schema, detected from `keyDeserializationClass`
+- ValueSchema: the schema, detected from `valueDeserializationClass`
```
```suggestion
- ValueSchema: the schema detected from `valueDeserializationClass`
```

Anonymitaet commented on site2/docs/io-kafka-source.md:
```diff
@@ -17,19 +17,48 @@ The configuration of the Kafka source connector has the following properties.
+Topic compaction and partition routing use the Pulsar key, that contains the Kafka key, and so they are driven by the same value that you have on Kafka.
+
+When you consume data from Pulsar topics, you can use the `KeyValue` schema. In this way, you can decode the data properly.
+In case you want to access the raw key you can use the Message#getKeyBytes() API.
```
```suggestion
If you want to access the raw key, you can use the `Message#getKeyBytes()` API.
```

Anonymitaet commented:
> @Jennifer88huang @Anonymitaet I have addressed your comments and tested again the rendering on my local env.
> PTAL
> 
> it is very slow to generate the website, I am using
> 
> ```
> yarn build
> yarn start
> ```

@eolivelli yes it works slow but they are the correct commands. We'd better preview the changes on our local environments and [check the broken links (if any)](https://github.com/apache/pulsar/tree/master/site2#check) before sending a doc PR.

eolivelli commented:
@Anonymitaet thanks again for your suggestions. I have updated the document
